src/modules/uORB/uORBDevices.cpp

```diff
@@ -148,8 +148,13 @@ uORB::DeviceNode::open(device::file_t *filp)
 			return -ENOMEM;
 		}
 
-		/* default to no pending update */
-		sd->generation = _generation;
+		/* If queue size >1, allow the subscriber to read the data in the queue. Otherwise, assume subscriber is up to date.*/
```

Comment:
Just as a note: This was discussed quite a bit with @dagar. It makes sure that if uORB messages of a certain type were published BEFORE the corresponding subscriber goes online (as is the case with commander, which starts way after the drivers that potentially publish subsystem_info messages start), these messages are still received by that subscriber IFF the message queue is larger than 1.

Reply:
I don't see any negative side-effects, but I don't like the condition on the queue size. It should be the same for all cases, but it would be too much of a change for the 1.8 release.

Reply:
Yes, the condition is there specifically to guarantee that all non-queued messages (which is the majority of messages) are absolutely unaffected such that this can be included in release 1.8.